src/entities/products-context-object.ts

```diff
@@ -0,0 +1,9 @@
+import { SetStateAction } from "react";
+import Product from "./product";
+
+export interface ProductsContextObject {
+  contextProducts: Product[];
+  refetchProducts: (data: SetStateAction<Product[]>) => void;
```

Comment:
I really don't think that you should use refetchProducts. Is this one only used for the delete functionality?

Reply:
For the moment yes, it is only used for the delete functionality. However, I may also use it for adding or updating products which I have to implement next.

I am aware that the current implementation of this is kind of "dirty", but I am sure things will get cleaner with a state management library.

I am open to suggestions and really curios about how I can improve this part.